manager/orchestrator/global.go

```diff
@@ -227,23 +239,40 @@ func (g *GlobalOrchestrator) reconcileOneService(ctx context.Context, service *a
 
 	_, err = g.store.Batch(func(batch *store.Batch) error {
 		var updateTasks []slot
-		for nodeID := range g.nodes {
+		for nodeID, node := range g.nodes {
+			meetsConstraints := constraint.NodeMatches(service.constraints, node)
 			ntasks := nodeTasks[nodeID]
+			delete(nodeTasks, nodeID)
+
 			// if restart policy considers this node has finished its task
 			// it should remove all running tasks
-			if _, exists := nodeCompleted[nodeID]; exists {
-				g.removeTasks(ctx, batch, service, ntasks)
-				return nil
+			if _, exists := nodeCompleted[nodeID]; exists || !meetsConstraints {
+				g.removeTasks(ctx, batch, ntasks)
+				continue
 			}
+
+			if node.Spec.Availability == api.NodeAvailabilityPause {
+				// the node is paused, so we won't add or update
```

Comment:
Paused means it doesn't accept new tasks, so I think a paused node should still kill the task if the service is deleted. That's how it works for replicated services.